### ISSUE TITLE
respect ID: link format in .org files

### ISSUE THREAD

**Issue by robby4769** [BBazard/obsidian-orgmode-cm6#6]
Hi there, thanks for making such a great plugin! 

As discussed on the obsidian forum, I'd love to see this plugin format and obey the "ID:" link format in org-mode. 

discussion in orgmode manual [here](https://orgmode.org/manual/Handling-Links.html)

and two screenshots (link display toggled on and off in emacs)
<img width="765" alt="image" src="https://github.com/BBazard/obsidian-orgmode-cm6/assets/13719668/0da5a606-54a4-43ec-a602-51c3fb662894">

<img width="765" alt="image" src="https://github.com/BBazard/obsidian-orgmode-cm6/assets/13719668/8abb8b79-4a9a-4a53-a726-29ab0ff4d6e0">

**Comment by BBazard** [comment]
It is a great feature I agree.

Linking to an ID inside the file in one thing, but linking to an ID in another file is more complicated.
It implies to **parse all the files** in the vault, the plugin currently only parse one file at a time.

My understanding is that emacs can only resolve ID links if you have the buffer of the target file open. Is that how you use it?

**Comment by robby4769** [comment]
No emacs (orgmode) will resolve an ID link anywhere in org-directory (and subdirectories if configured appropriately), whether or not it’s in a file that’s currently in a buffer.

I believe org-roam further enhances this by keeping a table of links and backlinks in a sqlite database, and updating the table every time a new link is created.

I agree; I don’t see any way to do it without some kind of either pre-processing or on-demand search. I imagine that’s how obsidian’s linking to nodes works as well?

in the short run, at least formatting the links, even if resolving doesn’t work, would be valuable. Especially with the auto org-generated uuid link style, my notes are pretty unreadable. 
![IMG_6564](https://github.com/BBazard/obsidian-orgmode-cm6/assets/13719668/2869a6dc-b8bb-419e-b769-c06b2bcc2043)

**Comment by BBazard** [comment]
Implemented in 2.3.0

I went with an on-demand search.
When clicking on an `:ID:` link, the plugin will open all the .org files of the vault, parse them and look for a matching id.
It works fine for a few files but it's possible it might take a long time with a large vault of org files, I didn't test it.
Obsidian has a caching mechanism to not have to read the files from disk every time so that's already a good optimization.
